api/filters/checkBrokerCredentials.go

```diff
@@ -0,0 +1,47 @@
+package filters
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	CheckBrokerCredentialsFilterName = "CheckBrokerCredentialsFilter"
+	credentialsPath                  = "credentials.basic.%s"
+)
+
+// CheckBrokerCredentialsFilter checks patch request for the broker basic credentials
+type CheckBrokerCredentialsFilter struct {
+}
+
+func (*CheckBrokerCredentialsFilter) Name() string {
+	return CheckBrokerCredentialsFilterName
+}
+
+func (*CheckBrokerCredentialsFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
+	fields := gjson.GetManyBytes(req.Body, "broker_url", fmt.Sprintf(credentialsPath, "username"), fmt.Sprintf(credentialsPath, "password"))
+
+	if fields[0].Exists() && !fields[1].Exists() && !fields[2].Exists() {
```

Comment:
With check like this if for example one provides URL and Username the respective expression will be:
`true && false && ...` Therefore the evaluation would stop when it sees that an username is provided even tough that password may be missing.
Something like `fields[0].Exists() && !(fields[1].Exists && fields[2].Exists)`